### PR TITLE
feat(CollisionSystem): add first version of collision system

### PR DESCRIPTION
**Pull request by sgalluz**
## 📋 Description
This pull request introduces a comprehensive collision system to the ECS (Entity Component System) framework, enabling entities to detect and resolve collisions with different physical responses (static, bounce, or none). It adds new components, collision resolution logic, and extensive test coverage to ensure correctness. The changes are grouped below by theme.

### Collision System Implementation

* Added a new `CollisionSystem` class that detects collisions between entities with `Position` and `BoxCollider` components, marks them as colliding, and dispatches resolution based on their collision response type.
* Introduced the `BoxCollider` component and the `CollisionResponse` enum to define collision behavior (`NONE`, `STATIC`, `BOUNCE`).

### Collision Resolution Logic

* Implemented the `CollisionResolver` interface and two concrete resolvers: `StaticResolver` (pushes entities out of static objects) and `BounceResolver` (makes entities bounce off each other by swapping velocities and separating them). [[1]](diffhunk://#diff-bd7b754e88cb9a465ea9f7d5dfe56a259d7db506b3eeaaf2c9dfbe8ecf5675ebR1-R7) [[2]](diffhunk://#diff-e15d47bf44b1c4479bbfa6a662de746b7722b82bdf0fe8b93c689c1678dbb1ecR1-R24) [[3]](diffhunk://#diff-2f47d8eb2d6fd478ddebf30739cf37c2d4d2e8d274c5373baf30638c0287944fR1-R30)
* Added the `CollisionManifold` data class to encapsulate collision details (overlap and direction).

### ECS Component and Entity Improvements

* Refactored the `Entity` class for improved component access: made the `components` map internal, simplified the `get` and `has` methods, and made `has` a reified inline function for type safety and convenience. [[1]](diffhunk://#diff-227864cf7aebbcc2bcf024360b2d48240a12436d237f8cf7a78387f577ea0875L10-R11) [[2]](diffhunk://#diff-227864cf7aebbcc2bcf024360b2d48240a12436d237f8cf7a78387f577ea0875L21-R29)

### Rendering Enhancements

* Updated the `ShapeRenderSystem` to visually indicate collisions by rendering entities in red when their `BoxCollider.isColliding` is true.

### Test Coverage

* Added extensive unit tests for the collision system (`CollisionSystemTest`) and bounce resolver (`BounceResolverTest`), covering detection, resolution, edge cases, and correct physical responses. [[1]](diffhunk://#diff-93542f7b9b5209dd95f28c3173a96a7a5acaf0cb43e676270d04f102356442e4R1-R257) [[2]](diffhunk://#diff-a47e0585d3fd4d945049349127cc91f3562a9358805f94cc0e44d2226e48d4a8R1-R155)

---

## 🔗 Related Issue
Not available

---

## 🚀 Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Tooling / Build system

---

## 🧪 Testing
How have you tested your changes?
- [x] Unit tests
- [ ] Manual testing (describe below)
- [ ] Not tested (explain why)

**Details**: Not needed

---

## ⚠️ Pre-Alpha Considerations

- [x] I am aware that K2D APIs are unstable
- [x] I kept the change minimal and focused
- [x] I avoided unnecessary public API exposure

---

## 📝 Checklist

- [x] Code follows the project style
- [x] New logic is documented (if applicable)
- [x] No existing tests are broken
- [x] I am willing to iterate based on feedback

### CODE REVIEW

File: engine/src/main/kotlin/io/sgalluz/k2d/ecs/systems/collision/CollisionData.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.sgalluz.k2d.ecs.systems.collision
+
+data class CollisionData(
+    val overlapX: Float,
+    val overlapY: Float,
+    val diffX: Float,
+    val diffY: Float
+)
```
Comment (sgalluz) [inline engine/src/main/kotlin/io/sgalluz/k2d/ecs/systems/collision/CollisionData.kt]: For improving the semantics, this could become something like:
```suggestion
data class CollisionManifold(
    val penetrationX: Float,
    val penetrationY: Float,
    val deltaX: Float,
    val deltaY: Float
)

```

File: engine/src/main/kotlin/io/sgalluz/k2d/ecs/systems/collision/BounceResolver.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import io.sgalluz.k2d.ecs.Entity
+import io.sgalluz.k2d.ecs.Position
+import io.sgalluz.k2d.ecs.Velocity
+
+class BounceResolver : CollisionResolver {
+    override fun resolve(e1: Entity, e2: Entity, manifold: CollisionManifold) {
+        val position1 = e1.get<Position>()!!
+        val velocity1 = e1.get<Velocity>()!!
+        val position2 = e2.get<Position>()!!
+        val velocity2 = e2.get<Velocity>()!!
```
[the remark this answers is not in the document]
Comment (sgalluz) [reply engine/src/main/kotlin/io/sgalluz/k2d/ecs/systems/collision/BounceResolver.kt]: The existence of Position and Velocity for entities is already checked in the BounceResolver caller so we can infer that these values will be valid
